### PR TITLE
`Modeling exercises`: Allow users to expand the modeling editor

### PR DESCRIPTION
**Pull request by iozbeyli**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest2.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
The modeling editors were horizontally expandable but not vertically. This PR brings the functionality.

### Description
Makes modeling editors in submission and assessment pages vertically expandable.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration
3. Create a modeling exercise
4. Log in as student and go to exercise
5. Expand the editor as seen in the screenshots
6. Submit your solution
7. Log in as tutor
8. Go to assessment of the submission 
9. Expand the editor as seen in the screenshots
10. Profit

### Review Progress

- Code Review
  - [x] Review 1
  - [x] Review 2
- Manual Tests
  - [x] Test 1
  - [x] Test 2

### Screenshots

https://user-images.githubusercontent.com/16825065/130885298-9a5e7c58-8072-4024-9513-208decd7b955.mp4

https://user-images.githubusercontent.com/16825065/130885346-075ffb28-c0aa-4809-82c0-76f6bb5dc060.mp4

### CODE REVIEW

File: src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { HttpErrorResponse } from '@angular/common/http';
-import { Component, HostListener, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { AfterViewInit, Component, HostListener, OnDestroy, OnInit, ViewChild } from '@angular/core';
```
Comment (ndricimrr) [inline src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts:2]: one unused import warning leftover.

File: src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.ts [hunk 2]
```diff
@@ -72,11 +72,11 @@ export class ModelingAssessmentComponent implements AfterViewInit, OnDestroy, On
...
             interact('.resizable')
                 .resizable({
-                    edges: { left: false, right: '.draggable-right', bottom: false, top: false },
+                    edges: { left: false, right: this.resizeOptions.initialWidth && '.draggable-right', bottom: this.resizeOptions.verticalResize, top: false },
                     modifiers: [
                         interact.modifiers!.restrictSize({
-                            min: { width: 15, height: 0 },
-                            max: { width: this.resizeOptions.maxWidth ? this.resizeOptions.maxWidth : 2500, height: 2000 },
+                            min: { width: 15, height: 1000 },
+                            max: { width: this.resizeOptions.maxWidth ? this.resizeOptions.maxWidth : 2500, height: 20000 },
```
Comment (ndricimrr) [inline src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.ts:79]: These seem like very magical numbers. Could we maybe have it in a constant instead 🤔 ?

File: src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts [hunk 3]
```diff
@@ -61,11 +61,11 @@ export class ModelingEditorComponent implements AfterViewInit, OnDestroy, OnChan
...
             interact('.resizable')
                 .resizable({
-                    edges: { left: false, right: '.draggable-right', bottom: false, top: false },
+                    edges: { left: false, right: this.resizeOptions.initialWidth && '.draggable-right', bottom: this.resizeOptions.verticalResize, top: false },
                     modifiers: [
                         interact.modifiers!.restrictSize({
-                            min: { width: 15, height: 0 },
-                            max: { width: this.resizeOptions.maxWidth ? this.resizeOptions.maxWidth : 2500, height: 2000 },
+                            min: { width: 15, height: 1000 },
+                            max: { width: this.resizeOptions.maxWidth ? this.resizeOptions.maxWidth : 2500, height: 20000 },
```
Comment (ndricimrr) [inline src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts:68]: could we maybe have the same thing here

**Review by ndricimrr — COMMENTED** [review]
Code changes look good 👍 . Left some minor comments

**Review by argertboja — COMMENTED** [review]
Technically everything works fine, however, I would suggest that the horizontal moving icon should be shown in the middle of the editor as it can be understood easier what it is used for and where it can be found :) I have also attached an image to show what I mean :)

No other issues were found while testing on TS1 👍🏼 Nice job 😄 
melih

**Comment by iozbeyli** [comment]
> Technically everything works fine, however, I would suggest that the horizontal moving icon should be shown in the middle of the editor as it can be understood easier what it is used for and where it can be found :) I have also attached an image to show what I mean :)
> 
> No other issues were found while testing on TS1 👍🏼 Nice job smile
> melih

That makes sense but would require a change in Apollon so I will politely decline this suggestion :smile:

File: src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export class ModelingEditorComponent implements AfterViewInit, OnDestroy, OnChan
     @Input()
     readOnly = false;
     @Input()
-    resizeOptions: { initialWidth: string; maxWidth?: number };
+    resizeOptions: { initialWidth?: string; maxWidth?: number; verticalResize?: boolean };
```
Comment (Olimbia23) [inline src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts:27]: Is there a reason why we don't have consistent flags for both vertical and horizontal resize?

**Review by Olimbia23 — APPROVED** [review]
Nice work @iozbeyli.
I looked at the code and change look good, I just have a small question about consistency of horizontal and vertical resize flags. Approve 👍

**Review by ndricimrr — COMMENTED** [review]
Tested again in TS1 after changes. Expanding seems to work well. 

I noticed a slight change that is different from before. The outline of the UML canvas is not in same width with upper blue part: 
cc0111

File: src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.scss [hunk 5]
```diff
@@ -19,7 +18,8 @@ $draggable-width: 15px;
 
 .resizable {
     display: flex;
-    .draggable-right {
+    .draggable-right,
```
Comment (ndricimrr) [inline src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.scss:21]: codacy issue

**Pull request merged**